apps/api/src/app/events/usecases/cancel-delayed/cancel-delayed.usecase.ts

```diff
@@ -1,33 +1,32 @@
 import { Injectable } from '@nestjs/common';
 import { JobStatusEnum, JobRepository } from '@novu/dal';
 import { CancelDelayedCommand } from './cancel-delayed.command';
-
+import { WorkflowQueueService } from '../../services/workflow-queue/workflow.queue.service';
 @Injectable()
 export class CancelDelayed {
-  constructor(private jobRepository: JobRepository) {}
+  constructor(private jobRepository: JobRepository, private workflowQueueService: WorkflowQueueService) {}
 
   public async execute(command: CancelDelayedCommand): Promise<boolean> {
     const job = await this.jobRepository.findOne({
       _environmentId: command.environmentId,
       transactionId: command.transactionId,
-      status: JobStatusEnum.DELAYED,
+      status: JobStatusEnum.QUEUED,
+      delay: { $gt: 0 },
```

Comment:
My thinking is delay should be part of the step/job(like an attribute), you take up any queue system/calendar, delay is part of event/job, there will not be any separate job for delay. I believe majority of our user perception also same.. Internally if we need to have rigid check we can move this to metadata(I have some ideas how to consolidate digest and delays, we can discuss later), coming to retrocompatible, this is fine with version upgrades, as part of this PR we are not deleting the delay jobs from DB, in DB still we have delay jobs like old way, while running we are merging delay jobs in to next active job and adding this delay in to next active job, so it works same like before(only in detail execution log instead of two steps we will have only one entry)

Reply:
Is a delay not a digest with a random digest key?

Reply:
@SebastianStehle based on our current assumption of digest job, delay job is different, but you are right, delay is like a digest job that will have unique random digest key and always digest one message, I consider even regular jobs are digest jobs with 1 message digested, I don't see why why we need to have different types of logic, delays and digests can be attributes of any step/job. if users does not enable digest then it becomes regular job(with digest count 1), same with delay...